storage/grpc_client.go

```diff
@@ -955,37 +958,44 @@ func (c *grpcStorageClient) RewriteObject(ctx context.Context, req *rewriteObjec
 	return r, nil
 }
 
-// bytesCodec is a grpc codec which permits receiving messages as either
-// protobuf messages, or as raw []bytes.
-type bytesCodec struct {
-	encoding.Codec
+type bytesCodecV2 struct {
+	encoding.CodecV2
 }
 
-func (bytesCodec) Marshal(v any) ([]byte, error) {
+// Marshal is used to encode messages to send for bytesCodecV2. Since we are only
+// using this to send ReadObjectRequest messages we don't need to recycle buffers
+// here.
+func (bytesCodecV2) Marshal(v any) (mem.BufferSlice, error) {
```

Comment:
OR, embed `encoding.CodecV2` and populate it with `encoding.GetCodec("proto")`, and don't override this method at all.

Reply:
Any particular reason to do this? or just to keep up with any changes to encoding.CodecV2's implementation?

Reply:
Yeah, in theory we could make improvements over there, and this would pick those up.  Definitely an optional thing to do, though.

Reply:
I'll leave this as is for now, can update this later.